src/Features/LanguageServer/Protocol/Handler/IRequestHandlerMetadata.cs

```diff
@@ -14,6 +16,11 @@ internal interface IRequestHandlerMetadata
         /// <summary>
         /// Name of the language for LSP method to handle (optional).
         /// </summary>
-        string LanguageName { get; }
+        string? LanguageName { get; }
+
+        /// <summary>
+        /// Whether or not handling this method results in changes to the current solution state
```

Comment:
i would add more information here.  i.e. what the implications of this are.  i.e. that mutation operations will be processed in a serialized fashion.  and non-mutating requests will operate on the last result of the prior mutating requset.